### PR TITLE
Added GitHub workflows

### PR DESCRIPTION
**Pull request by Fuzznimp**
## Description

This PR aims at running the lint and the test suit in case a file has been changed in the respective package, done through the `path` key.

I also removed the env vars I don't want to source the container.

## Motivation and Context

This will prevent forgetting to run the linter

## How Has This Been Tested?

I tested it by changing some files in the different repositories to see if the workflows were triggered.

## Types of changes

- [ ] Chore (non-breaking change which refactors / improves the existing code base)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] My change requires a change to a package version.
- [ ] I have updated the `package.json` version accordingly.
- [ ] I have updated the `CHANGELOG.md` version accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-hf8ukh [ci] - Run `connect-js` tests in CI](https://app.clickup.com/t/hf8ukh)

File: .github/workflows/lint-and-test-management.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        path: |
+          ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          **/management/node_modules
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    
+    - name: dependencies
+      if: steps.yarn-cache.outputs.cache-hit != 'true'
+      run: cd management/ && yarn install
```
Comment (Fenntasy) [inline .github/workflows/lint-and-test-management.yml:57]:
```suggestion
    - name: dependencies
      if: steps.yarn-cache.outputs.cache-hit != 'true'
      run: yarn install
      working-directory: management
```

source: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun

File: .github/workflows/lint-and-test-management.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    - name: tests
+      run: cd management/ && yarn test
+      env:
+        PROFILE_MOCK_SERVER_URL: "http://localhost:29712"
```
Comment (Fenntasy) [inline .github/workflows/lint-and-test-management.yml:62]:
```suggestion
    - name: tests
      run: yarn test
      working-directory: management
      env:
        PROFILE_MOCK_SERVER_URL: "http://localhost:29712"
```

**Review by Fenntasy — COMMENTED** [review]
Do you know why there's only the `client-linter` and `client-tests` in the Check tab?

**Comment by Fuzznimp** [comment]
> Do you know why there's only the `client-linter` and `client-tests` in the Check tab?

Because code were only changed there (PR discription 😁)

**Comment by Fenntasy** [comment]
> Because code were only changed there (PR discription 😁)

Ooooooh 👍

**Review by Fenntasy — APPROVED** [review]

**Pull request merged**